app-typescript/components/Switch.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import classNames from 'classnames';
+
+interface IProps {
+    value?: boolean;
```

Comment:
make value required

Reply:
Why? If it is required you will need to specify value all the time. If it is not required, you can just go with default behaviour (false). It's cleaner if we go with natural approach.

Reply:
But yea, if we specify model inside value, then it will automatically be changed when we change props value. Changing...

Reply:
Then we can make onChange not required in this case

Reply:
When you use the component, you'll have to provide `onChange` function, which will update the state in another component. It's already in the state elsewhere, so it's not useful having it twice.

When would you have `onChange` optional? What's the use of the switch if you can't change the value?

Reply:
But `this.props` are readonly. How to change internal behaviour (adding class) if I don't have internal state? I will need to have internal state so I can set checked true or false and add corresponding class. Why I need to set `value` and `onChange` both required? `value` can go with default behaviour of false, and onChange will update state in another component.

Reply:
I think I get what you want to say. This will be handled outside, `this.props.value` will be changed outside of this component

Reply:
That's right. You will always have the newest value in `this.props`. See https://reactjs.org/docs/forms.html#controlled-components